wit/deps/golem/golem-host.wit

```diff
@@ -53,6 +53,98 @@ interface host {
         smart
     }
 
+    enum filter-comparator {
+        equal,
+        not-equal,
+        greater-equal,
+        greater,
+        less-equal,
+        less
+    }
+
+    enum string-filter-comparator {
+        equal,
+        not-equal,
+        like,
+        not-like
+    }
+
+    enum worker-status {
+        /// The worker is running an invoked function
+        running,
+        /// The worker is ready to run an invoked function
+        idle,
+        /// An invocation is active but waiting for something (sleeping, waiting for a promise)
+        suspended,
+        /// The last invocation was interrupted but will be resumed
+        interrupted,
+        /// The last invocation failed and a retry was scheduled
+        retrying,
+        /// The last invocation failed and the worker can no longer be used
+        failed,
+        /// The worker exited after a successful invocation and can no longer be invoked
+        exited,
+    }
+
+    record worker-name-filter {
+        comparator: string-filter-comparator,
+        value: string
+    }
+
+    record worker-status-filter {
+        comparator: filter-comparator,
+        value: worker-status
+    }
+
+    record worker-version-filter {
+        comparator: filter-comparator,
+        value: s32
+    }
+
+    record worker-created-at-filter {
+        comparator: filter-comparator,
+        value: u64
+    }
+
+    record worker-env-filter {
+        name: string,
+        comparator: string-filter-comparator,
+        value: string
+    }
+
+    variant worker-property-filter {
+        name(worker-name-filter),
+        status(worker-status-filter),
+        version(worker-version-filter),
+        created-at(worker-created-at-filter),
+        env(worker-env-filter)
+    }
+
+    variant worker-filter {
+        or(list<worker-property-filter>),
+        and(list<worker-property-filter>)
+    }
+
+    record worker-metadata {
+        worker-id: worker-id,
+        args: list<string>,
+        env: list<tuple<string, string>>,
+        status: worker-status,
+        template-version: s32,
+        retry-count: s32
+    }
+
+    record workers-metadata-response {
+        workers: list<worker-metadata>,
+        cursor: option<u64>
+    }
+
+    resource get-workers {
+        constructor(template-id: template-id, filter: worker-filter, count: u64, precise: bool);
```

Comment:
Not sure we need to expose `count` - could be decided by the implementation. Especially that once WASI Preview3 is done, we can convert this whole thing to a `stream`. 

But if you think there is a good reason to have it then let's call it `page-size` or something that better describes what it is.